frontend/packages/console-app/src/components/guided-tours/GuidedTourDrawer.tsx

```diff
@@ -0,0 +1,63 @@
+import * as React from 'react';
+import {
+  Drawer,
+  DrawerPanelContent,
+  DrawerContent,
+  DrawerPanelBody,
+  DrawerHead,
+  DrawerActions,
+  DrawerCloseButton,
+  Alert,
+  Title,
+} from '@patternfly/react-core';
+import './GuidedTourDrawer.scss';
+
+type GuidedTourDrawerProps = {
+  expanded: boolean;
+  title: string;
+  tourDuration: number;
+  onClose: () => void;
+  guidedTourContent: JSX.Element;
+  children: React.ReactNode;
+  isComplete?: boolean;
+};
+
+const GuidedTourDrawer: React.FC<GuidedTourDrawerProps> = ({
+  expanded,
+  title,
+  tourDuration,
+  onClose,
+  guidedTourContent,
+  children,
+  isComplete,
+}) => {
+  const panelContent = (
+    <DrawerPanelContent className="co-guided-tour-drawer__panel-content">
+      <DrawerHead>
+        <div>
+          <Title size="xl" style={{ display: 'inline' }}>
+            {title}
+          </Title>
+          <span className="text-muted co-guided-tour-drawer__subtitle">
+            {`${tourDuration} minutes`}
+          </span>
+        </div>
+        <DrawerActions>
+          <DrawerCloseButton onClick={onClose} />
+        </DrawerActions>
+        {isComplete && (
+          <Alert variant="success" isInline title="This tour has already been completed" />
+        )}
+      </DrawerHead>
+      <DrawerPanelBody>{guidedTourContent}</DrawerPanelBody>
+    </DrawerPanelContent>
+  );
+
+  return (
+    <Drawer isExpanded={expanded} isInline>
+      <DrawerContent panelContent={panelContent}>{children}</DrawerContent>
+    </Drawer>
+  );
+};
+
+export default GuidedTourDrawer;
```

Comment:
We can indeed connect to redux in the component itself, but i chose to keep the drawer component separate from the store since I saw few components where this convention has been followed and it makes testing(if any) more convenient.